rust/datafusion/examples/database_sql.rs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::util::pretty;
+
+use datafusion::error::Result;
+use datafusion::prelude::*;
+
+/// This example demonstrates executing a simple query against a SQL database and
+/// fetching results.
+/// The example assumes that one has a table called `nyc_yellow_taxi` in a postgresql
+/// database.
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Please note, the example is to demonstrate functionality, and it reads in
```

Comment:
@andygrove @alamb @jorgecarleitao here's an example where I've glued together the SQL connection.
In this case, I read ~2 million records from the DB, and then apply a query via DataFusion.

In this state, we wouldn't have any project pushdown, or even the ability to filter, as it seems we'd need the ability to convert a physical plan to a SQL query. I don't think `sqlparser` supports this now, but it'd be very cool.